services/comps/block/src/request_queue.rs

```diff
@@ -7,24 +7,113 @@ use super::{
     id::Sid,
 };
 
-/// Represents the software staging queue for the `BioRequest` objects.
-pub trait BioRequestQueue {
+use aster_frame::sync::{Mutex, WaitQueue};
+
+/// A simple block I/O request queue with a single queue.
```

Comment:
This sentence can be boiled down to "a queue with a queue". This expression is illogical. Consider rewriting this way.

```
/// A simple block I/O request queue backed by one internal FIFO queue.
```

In addition, the Rust doc should mention that the queue support merging the new request with the front request.

Reply:
Done